param_passing/proposal.md

```diff
@@ -0,0 +1,388 @@
+# Proposal for parameter passing
+
+* **Authors**:
+Jenn Nguyen `<jenn@openrobotics.org>`,
+Addisu Taddese `<addisu@openrobotics.org>`
+* **Status**: Draft
+* **SDFormat Version**: 1.7
+* **`libsdformat` Version**: 10
+
+## Introduction
+
+**Purpose statement**
+
+This proposal suggests extending the [composition behavior in SDFormat 1.7](
+  http://sdformat.org/tutorials?tut=composition#including-a-model)
+(i.e., using the `<include>` tag) to pass additional arguments to SDFormat files
+which will allow a user to send custom data into a model file and reduce the
+need for model file duplication.
+
+**Background**
+
+Currently, through the `<include>` (i.e., `//include`) tag, model parameters
+such as `//model/name` and `//model/pose` may be overwritten but there are
+cases when other parameters are desired to be updated as well. For example, if
+several robots have the same base but different sensor configurations then
+several model files need to be created to compose all the robot variations.
+This is an issue because these model files are most likely copies of each other
+with certain parameters changed.
+
+**Conclusion**
+
+The aim of adding additional parameter support is to avoid this duplication,
+where a model file contains all configurations with default parameters and a
+user may update and/or override certain parameters as desired in a model or
+world file when using the `//include` tag.
+
+## Document summary
+
+* *Proposed changes*: details the proposed design and the actions available to
+users
+* *Example*: provides an example of how this proposed approach can be utilized
+
+
+## Motivation
+
+The motivation behind this proposal is to avoid duplication when constructing
```

Comment:
Are there examples of using Xacro (as Grant alluded to)?

We have some, but may take some time to suss out of our internal repo at TRI.
Perhaps there are other already-open-sourced models that you can reference, perhaps even those that use URDF?

Reply:
This will only be implemented for SDF files and not Xacro or URDF